### DISCUSSION TITLE
Productionization (k8s configs?)

### DISCUSSION THREAD

**Discussion by ellenhp** [headwaymaps/headway#39] [Ideas]
I just wanted to start a thread of discussion around kubernetes configs. I don't know much about kubernetes but I just want to lay out a few requirements for whatever we end up implementing (and collect more requirements from others)

1. It shouldn't be too difficult to deploy a few different Headway instances for a few different cities (does this mean we need deployment templates somehow?)
2. It should be easy to upload a new data volume and have it propagate to an existing cluster (by bringing up a new pod? I'm so bad at k8s)
3. For larger installations it would be really nice to let nominatim and maybe also photon cache the contents of their indexes in a volume specific to them.
4. Contention of the data volume and the photon/nominatim volumes during updates shouldn't be an issue (this would make zero-downtime deploys very hard).

That's all I can think of right now.

**Comment by 3nprob** [comment]
I'd like to add one more:

* It shouldn't be assumed that users are using either compose or k8s - headway containers should run properly out of the box regardless of orchestration. Outside of the k8s deployment config and potential helm chart, other parts of headway should not need to be aware of it. 

I mention this since I've experienced projects in the past that make such assumptions in initscripts and whatnot, which easily becomes a mess. So e.g. deploying for multiple cities being straightforward should not rely on using k8s.

**Reply by ellenhp** [reply]
Sounds reasonable.
